src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -1497,15 +1515,27 @@ private static void EmitLoadParamArg(ILGenerator il, int paramIndex, bool asAddr
                 {
                     var convertOpMethod = FirstConvertOperatorOrDefault(sourceTypeInfo, targetType, sourceType);
                     if (convertOpMethod != null)
-                        return EmitMethodCall(il, convertOpMethod);
+                    {
+                        if (!EmitMethodCall(il, convertOpMethod))
+                            return false;
+                        if (!shouldReturnValue)

```

Comment:
Could we add this check into `EmitMethodCall`?
I saw it repeating together at least 3 times.

Reply:
changed